### PR TITLE
[Spark-17745][ml][PySpark] update NB python api - add weight col parameter

### PR DESCRIPTION
**Pull request by WeichenXu123**
## What changes were proposed in this pull request?

update python api for NaiveBayes: add weight col parameter.
## How was this patch tested?

doctests added.

### CODE REVIEW

**Comment by sethah** [comment]
Would you mind changing the title to reflect the actual change? Something to indicate that you are adding weight column param to Python? Thanks!

**Comment by WeichenXu123** [comment]
@sethah OK. and I'm checking whether there is something else need to update...

**Comment by sethah** [comment]
We should add weights to the doctests to demonstrate them and make sure they're working.

File: python/pyspark/ml/classification.py [hunk 1]
```diff
@@ -1032,6 +1032,16 @@ class NaiveBayes(JavaEstimator, HasFeaturesCol, HasLabelCol, HasPredictionCol, H
     >>> result = model3.transform(test0).head()
     >>> result.prediction
     0.0
+    >>> dfWithWeight = spark.createDataFrame([
```
Comment (sethah) [inline python/pyspark/ml/classification.py]: Sorry for the confusion, I was thinking we could just add weights to the existing example. I don't think an entirely separate example is necessary. What do you think?

Comment (yanboliang) [inline python/pyspark/ml/classification.py]: +1 @sethah

**Comment by sethah** [comment]
LGTM

**Comment by yanboliang** [comment]
LGTM2, merged into master. Thanks!

**Pull request closed without merging**